crates/burn-dataset/src/vision/image_folder.rs

```diff
@@ -76,7 +76,7 @@ pub enum Annotation {
 #[derive(Debug, Clone, PartialEq)]
 pub struct SegmentationMask {
     /// Segmentation mask.
-    pub mask: Vec<usize>,
+    pub mask: Vec<PixelDepth>,
```

Comment:
For segmentation masks, I don't think it makes as much sense to have them stored as `PixelDepth` since the image should have integer values to represent a class index. We could keep that as a vec of usize.